src/main/java/ru/yandex/practicum/filmorate/dao/impl/FilmDbStorage.java

```diff
@@ -148,30 +153,123 @@ public Optional<Film> update(Film film) {
+        String sqlQuery =
+                "DELETE FROM film " +
+                        "WHERE film_id = ?";
+
+        jdbcTemplate.update(sqlQuery, filmId);
+    }
+
+    @Override
+    public Collection<Film> getFilmRecommendations(int userId) throws EmptyResultDataAccessException {
+        String sqlQuery =
+                "WITH rec_user AS " +
+                        "(SELECT t2.user_id " +
+                        "FROM film_like t1 " +
+                        "INNER JOIN film_like t2 ON t1.film_id = t2.film_id " +
+                        "AND t2.user_id <> t1.user_id " +
+                        "WHERE t1.user_id = ? " +
+                        "GROUP BY t2.user_id " +
+                        "ORDER BY COUNT(t2.film_id) DESC " +
+                        "LIMIT 1) " +
+
+                        "SELECT rec.film_id " +
+                        "FROM film_like rec " +
+                        "INNER JOIN rec_user ON rec.user_id = rec_user.user_id " +
+                        "LEFT JOIN film_like base ON rec.film_id = base.film_id " +
+                        "AND base.user_id = ? " +
+                        "WHERE 1=1 " +
+                        "AND base.film_id IS NULL";
+
+        return jdbcTemplate.queryForList(sqlQuery,
+                        Integer.class,
+                        userId,
+                        userId)
+                .stream()
+                .map(this::getById)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public Collection<Film> getCommonFilms(int userId, int friendId) {
+        String sqlQuery =
+                "SELECT film_id " +
+                        "FROM film_like " +
+                        "WHERE user_id = ? " +
+                        "INTERSECT " +
+                        "SELECT film_id " +
+                        "FROM film_like " +
+                        "WHERE user_id = ?";
+        return jdbcTemplate.queryForList(sqlQuery,
+                        Integer.class,
+                        userId,
+                        friendId)
+                .stream()
+                .map(this::getById)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toList());
+    }
+
     private Film makeFilm(ResultSet rs) throws SQLException {
         int filmId = rs.getInt("film_id");
         Film film = Film.builder()
-                .name(rs.getString("film_nm"))
-                .description(rs.getString("film_desc"))
+                .name(rs.getString("name"))
+                .description(rs.getString("description"))
                 .releaseDate(rs.getDate("release_dt").toLocalDate())
-                .duration(rs.getInt("duration_min"))
+                .duration(rs.getInt("duration"))
                 .mpa(mpaStorage.getById(rs.getInt("rating_id")).orElseGet(null))
```

Comment:
Лучше заменить на 
```suggestion
                .mpa(mpaStorage.getById(rs.getInt("rating_id")).orElse(null))
```
Так как `.orElseGet` `@NotNull`